app/services/resolver_cost_calculator.rb

```diff
@@ -0,0 +1,35 @@
+class ResolverCostCalculator
+  def initialize(source)
+    @source = source
+  end
+
+  def cost
+    @source.outcome == 'none' ? 0 : incurred_cost(@source)
+  end
+
+  private
+
+  def incurred_cost(source)
+    if source.is_a?(PartPayment)
+      if source.application.evidence_check.present?
+        fee(source) - amount_to_pay(source.application.evidence_check)
+      else
+        fee(source) - amount_to_pay(source.application)
+      end
+    else
+      fee(source)
+    end
+  end
```

Comment:
I'll have a look what I can do. This suggestion I find personally more confusing, because for example `incurred_cost != process_part_payment` which this method suggest. But I'll see what I can do tomorrow morning with fresh eyes.